StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp7/SpacingRules/SA1000CSharp7UnitTests.cs

```diff
@@ -228,15 +231,16 @@ public void TestMethod()
 }
 ";
 
-            DiagnosticResult[] expectedDiagnostics =
+            await new CSharpTest
             {
-                this.CSharpDiagnostic().WithArguments("foreach", string.Empty, "followed").WithLocation(7, 13),
-                this.CSharpDiagnostic().WithArguments("foreach", string.Empty, "followed").WithLocation(8, 13),
-            };
-
-            await this.VerifyCSharpDiagnosticAsync(testCode, expectedDiagnostics, CancellationToken.None).ConfigureAwait(false);
-            await this.VerifyCSharpDiagnosticAsync(fixedCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
-            await this.VerifyCSharpFixAsync(testCode, fixedCode).ConfigureAwait(false);
+                TestCode = testCode,
+                FixedCode = fixedCode,
+                ExpectedDiagnostics =
+                {
+                    Diagnostic().WithArguments("foreach", string.Empty, "followed").WithLocation(7, 13),
+                    Diagnostic().WithArguments("foreach", string.Empty, "followed").WithLocation(8, 13),
+                },
+            }.RunAsync(CancellationToken.None).ConfigureAwait(false);
```

Comment:
Async adds a bit of noise and ways to get things wrong to tests. Doing antipattern task.wait() and exposing as blocking is an alternative.

Reply:
➡️ This is unlikely to change for StyleCop Analyzers. Also, while the version eventually going to dotnet/roslyn-sdk is likely to continue to use the asynchronous approach (a more common practice in xUnit than in NUnit), two changes are likely to reduce the noise:

1. Making the `CancellationToken` optional
2. Not requiring `ConfigureAwait(false)`

Reply:
Making `CancellationToken` optional or removing it makes sense. I have never had a case where I needed to test cancellation.

Not requiring `ConfigureAwait` is not going to change much as caller will likely kneejerk writing it any way.

I'm not sure it is a more common practice when using nunit, async tests are async with NUnit too.
The question is if there is a real problem with doing `task.Wait()` inside the fixture. An alternative is to expose async overloads.

Reply:
```cs
[Test]
public async Task Test()
{
    await VerifyAsync(code).ConfigureAwait(false);
}
```
There are three points where a user can get ^ wrong.
1. Not making return type `async Task`
2. Forgetting `await`
3. Forgetting `ConfigureAwait`

I'm yet to see a deadlock from doing antipattern `Wait()` in asserts. Do you have a repro that  deadlocks a test?